SYCL/SubGroup/load_store.cpp

```diff
@@ -51,7 +46,9 @@ template <typename T, int N> void check(queue &Queue) {
           {L + max_sg_size * N}, cgh);
       cgh.parallel_for<sycl_subgr<T, N>>(NdRange, [=](nd_item<1> NdItem) {
         ONEAPI::sub_group SG = NdItem.get_sub_group();
-        if (SG.get_group_id().get(0) % N == 0) {
+        auto SGid = SG.get_group_id().get(0);
+        /* Avoid overlapping data ranges inside and between local groups */
+        if (SGid % N == 0 && (SGid + N) * SG.get_local_range()[0] <= L) {
```

Comment:
Please note that `get_local_range` can return different value for the last sub-group if local size is not evenly divisible by sub-group size.